graph/pkg/service/v0/groups.go

```diff
@@ -28,6 +30,89 @@ func (g Graph) GetGroups(w http.ResponseWriter, r *http.Request) {
 	render.JSON(w, r, &listResponse{Value: groups})
 }
 
+// PostGroup implements the Service interface.
+func (g Graph) PostGroup(w http.ResponseWriter, r *http.Request) {
+	grp := libregraph.NewGroup()
+	err := json.NewDecoder(r.Body).Decode(grp)
+	if err != nil {
+		errorcode.InvalidRequest.Render(w, r, http.StatusBadRequest, err.Error())
+		return
+	}
+
+	if isNilOrEmpty(grp.DisplayName) {
+		errorcode.InvalidRequest.Render(w, r, http.StatusBadRequest, "Missing Required Attribute")
+		return
+	}
+
+	// Disallow user-supplied IDs. It's supposed to be readonly. We're either
+	// generating them in the backend ourselves or rely on the Backend's
+	// storage (e.g. LDAP) to provide a unique ID.
+	if !isNilOrEmpty(grp.Id) {
+		errorcode.InvalidRequest.Render(w, r, http.StatusBadRequest, "group id is a read-only attribute")
+		return
+	}
+
+	if grp, err = g.identityBackend.CreateGroup(r.Context(), *grp); err != nil {
+		errorcode.GeneralException.Render(w, r, http.StatusInternalServerError, err.Error())
+		return
+	}
+
+	render.Status(r, http.StatusOK)
+	render.JSON(w, r, grp)
+}
+
+// PatchGroup implements the Service interface.
+func (g Graph) PatchGroup(w http.ResponseWriter, r *http.Request) {
+	g.logger.Debug().Msg("Calling PatchGroup")
+	groupID := chi.URLParam(r, "groupID")
+	groupID, err := url.PathUnescape(groupID)
+	if err != nil {
+		errorcode.InvalidRequest.Render(w, r, http.StatusBadRequest, "unescaping group id failed")
+	}
+
+	if groupID == "" {
+		errorcode.InvalidRequest.Render(w, r, http.StatusBadRequest, "missing group id")
+		return
+	}
+	changes := libregraph.NewGroup()
+	err = json.NewDecoder(r.Body).Decode(changes)
+	if err != nil {
+		errorcode.InvalidRequest.Render(w, r, http.StatusBadRequest, err.Error())
+		return
+	}
+
+	if memberRefs, ok := changes.GetMembersodataBindOk(); ok {
+		var memberIDs []string
+		for _, memberRef := range memberRefs {
+			memberType, id, err := g.parseMemberRef(memberRef)
+			if err != nil {
+				errorcode.InvalidRequest.Render(w, r, http.StatusInternalServerError, "Error parsing member@odata.bind values")
+				return
```

Comment:
Just to be clear: If I patch a group adding 1000 members and one of them is corrupted, I want the group not to be created at all. Is that what we want here?

Reply:
That's a very good question. I think it's the best thing we can do here. Ignoring invalid/non-existing users here and just returning success when some members are skipped sound somehow wrong to me.

But this reminds me that the MS Graph API limit the the Patch to a maximum of 20 members per request. I still need to add that. I would however prefer to track that via  an issue and add that in a separate PR.

Reply:
All good for me :+1: Just wanted to make sure it is not unintentional...

Reply:
BTW, I add the 20 members per Request limit now as well.